### ISSUE TITLE
Create a better examples and instead of using built-in demo window

### ISSUE THREAD

**Issue by swistakm** [swistakm/pyimgui#45]
As suggested by @ccanepa on Cocos2d mailing group:

> while the example does a good job showing the wide offering, it exposes something that is entirely built on the other side of the bind. An additional, much simpler but non trivial example built in the python side would help to grasp how to compose the window.

**Comment by stuaxo** [comment]
I recently did the React tutorial, a tick tac toe game, and was quite impressed.
I'm not going to suggest doing the same tutorial but explaining why it was good could may be useful.

In the tutorial, the developer builds the game, with roughly these steps -

 - Draw the board
 - Add simplest interaction (clicking to add an X or an O)
 - Add turns
 - Checking if there anyone has won
 - Then in the final stage a game-history / undo is added.

It starts in the most basic way, and at every state is implemented about as simply as possible, growing to something of medium complexity.

Unlike other tutorials that just teach you an API, this one seems to subtly teach you how to grow your app, showing the best place to keep state at each stage, and where to move that as functionality gets more complex.

**Comment by pconerly** [comment]
@swistakm I had the same though, I was thinking we should have a `imgui._py_show_demo_window()` that calls a pyimgui-implemented demo window. That'd be really helpful to expose what is and isn't yet available in the pyimgui API.

**Comment by learn-more** [comment]
The beginnings of this are now present:
https://github.com/pyimgui/pyimgui/blob/dev/version-2.0/doc/examples/testwindow.py

**Comment by pvallet** [comment]
Just leaving a comment here to mention my interest in pconerly's suggestion, a python-implemented version of show_demo_window seems like a must-have to me. Following the same pattern as imgui where the example is embedded in the source (instead of in a doc) is the cleanest in my opinion, as it allows showing everything that is supported by the library in a way that is very accessible to coders. 
If you don't remember how to use a certain widget, just call `imgui.show_demo_window()`, then select it and use "jump to source" in your IDE, and you have direct access to the examples, no need to find the right files.
In my opinion, `show_demo_window()` should be the python implementation, and the call to the C++ version, which would only be used to compare what is missing from the API, could be called something like `show_demo_window_native()` or something.

**Comment by KinoxKlark** [comment]
I agree with that. I believe some works have already been done by @billsix some time ago. Here: https://github.com/pyimgui/pyimgui/pull/206 and https://github.com/pyimgui/pyimgui/pull/223. 

It has been a long time since I checked, so I am unsure how many of the proposed functions are documented here.

**Comment by billsix** [comment]
Hey! Sorry for my delay in responding, yeah, I've made a bunch of the example code in Python instead of in c++.. I can start back up again soon adding more to the demos if people are interested.  

I got the demo to the point where was complete for what I needed it for, but I'm more than happy to make it as feature complete as possible if it will help other people out

**Comment by pvallet** [comment]
Hey! That would definitely be much appreciated, for the reasons mentioned above (examples serving as documentation). Also, once the pattern is established, it's easier to make sure that subsequent pyimgui bindings also provide an example in the window. 
I'm sorry I didn't have the time to look into it myself, but I'm already struggling to dedicate enough time to my main project that uses pyimgui. Basically, what I'm doing now is call the demo window, look into the name of the widget I want in C++, and then search for it on the pyimgui documentation page. Being able to look up the widget functions directly would make the whole thing so much easier

**Comment by billsix** [comment]
I committed the following Python file a few years ago, it should do a decent amount.  I can continue to uncomment some of the c++ sections and turn it into Python, but there's a lot there already

https://github.com/pyimgui/pyimgui/blob/master/doc/examples/testwindow.py
